### PR TITLE
Reduce file descriptor consumption

### PR DESCRIPTION
**Pull request by bsneed**
Reduces file descriptor usage by storing file handle rather than opening a new one with every write.

### CODE REVIEW

File: Sources/Segment/Utilities/Storage.swift [hunk 1]
```diff
@@ -241,23 +243,20 @@ extension Storage {
...
-                    handle.write(jsonData)
-                    handle.closeFile()
-                } else {
-                    assert(false, "Storage: Unable to convert event to json!")
+            let jsonString = event.toString()
+            if let jsonData = jsonString.data(using: .utf8) {
+                fileHandle?.seekToEndOfFile()
+                // prepare for the next entry
+                if newFile == false {
+                    fileHandle?.write(",".data(using: .utf8)!)
```
Comment (migs647) [inline Sources/Segment/Utilities/Storage.swift:251]: Do we absolutely know 100% of the time this explicit will work on all platforms? Makes me uneasy.

File: Sources/Segment/Utilities/Storage.swift [hunk 2]
```diff
@@ -241,23 +243,20 @@ extension Storage {
...
                 }
-            } catch {
-                assert(false, "Storage: failed to write event to \(storeFile), error: \(error)")
+                // write the data
+                fileHandle?.write(jsonData)
+                if #available(tvOS 13, *) {
```
Comment (migs647) [inline Sources/Segment/Utilities/Storage.swift:255]: Might be good to add a comment why we are doing this for 13 and above but supporting 11.

File: Sources/Segment/Utilities/Storage.swift (hunk 1 above)
Comment (bsneed) [reply Sources/Segment/Utilities/Storage.swift:251]: Why would it not work on all platforms??  It's almost identical to what it was before and tests are passing hunky dory.

Comment (migs647) [reply Sources/Segment/Utilities/Storage.swift:251]: I guess it just being a comma should be okay.

**Review by migs647 — APPROVED** [review]

**Pull request merged**